src/signTx.c

```diff
@@ -48,7 +48,13 @@ static inline void advanceStage()
 		ASSERT(ctx->currentInput == ctx->numInputs);
 		txHashBuilder_enterOutputs(&ctx->txHashBuilder);
 		ctx->stage = SIGN_STAGE_OUTPUTS;
-		break;
+
+		if (ctx->numOutputs > 0) {
```

Comment:
the CDDL specifies that the outputs key is obligatory: https://github.com/input-output-hk/cardano-ledger-specs/blob/6e30ecf73c9296bc320d71f3a6b7bfad3ce117e9/shelley/chain-and-ledger/shelley-spec-ledger-test/cddl-files/shelley.cddl#L53 - I guess this code results in omitting the key completely, right?

Reply:
We are always  calling `txHashBuilder_enterOutputs` (see a couple of lines above) which is responsible for this.